src/NanoMsg.h

```diff
@@ -0,0 +1,19 @@
+#include <nanomsg/nn.h>
+#include <string>
+#include <vector>
+/*
+ * class for encapsulating nanomsg zero copy allocation/deletion
+ */
+class NanoMsg {
+   typedef const std::vector<std::pair<void*, unsigned int>> PacketHashVector;
+public:
+   NanoMsg(PacketHashVector& data);
+   NanoMsg(void*& data);
+   NanoMsg(const std::string& data);
+   ~NanoMsg();
+   NanoMsg& operator=(const NanoMsg&) = delete;
+   NanoMsg(const NanoMsg&) = delete;
+   NanoMsg() = delete;
+   void* buffer;
```

Comment:
I think it's better to go through an accessor. That way you can (if you wanted to) put in CHECK and asserts for nullptr